build/bin/write-dockerfile.sh

```diff
@@ -214,6 +214,18 @@ EOF
         ;;
   esac
 esac
+
+case ${DOCKER_BUILDKIT-0} in
+    1)
+        # With buildkit we cannot retrieve failed builds.
+        # So we do not allow the main step of a build stage to fail.
+        # Instead we record the exit code in the file STATUS.
+        THEN_SAVE_STATUS='; echo $? > STATUS'
+        # ... and at the beginning of the next build stage,
+        # we check the status and exit with an error status.
+        CHECK_STATUS_THEN='STATUS=$(cat STATUS 2>/dev/null); case "$STATUS" in ""|0) ;; *) exit $STATUS;; esac; '
+esac
```

Comment:
Out of curiosity, how does this trick avoid the problem of "With buildkit we cannot retrieve failed builds."?

How is "failed build" different from "exit" with nonzero status?

Reply:
> Out of curiosity, how does this trick avoid the problem of "With buildkit we cannot retrieve failed builds."?

By having the next build stage fail at the beginning, instead of failing the current one.

Reply:
so that the previous stage remains in the container and the current one discarded...

OK.